waku/waku_store_sync/reconciliation.nim

```diff
@@ -0,0 +1,358 @@
+{.push raises: [].}
+
+import
+  std/sequtils,
+  stew/byteutils,
+  results,
+  chronicles,
+  chronos,
+  metrics,
+  libp2p/utility,
+  libp2p/protocols/protocol,
+  libp2p/stream/connection,
+  libp2p/crypto/crypto,
+  eth/p2p/discoveryv5/enr
+import
+  ../common/nimchronos,
+  ../common/protobuf,
+  ../common/paging,
+  ../waku_enr,
+  ../waku_core,
+  ../node/peer_manager/peer_manager,
+  ../waku_archive,
+  ./common,
+  ./codec,
+  ./storage/storage,
+  ./storage/seq_storage,
+  ./storage/range_processing,
+  ./protocols_metrics
+
+logScope:
+  topics = "waku reconciliation"
+
+const DefaultStorageCap = 50_000
+
+type SyncReconciliation* = ref object of LPProtocol
+  peerManager: PeerManager
+
+  wakuArchive: WakuArchive
+
+  storage: SyncStorage
+
+  # Receive IDs from transfer protocol for storage
+  idsRx: AsyncQueue[ID]
+
+  # Send Hashes to transfer protocol for reception
+  wantsTx: AsyncQueue[(PeerId, Fingerprint)]
+
+  # Send Hashes to transfer protocol for transmission
+  needsTx: AsyncQueue[(PeerId, Fingerprint)]
```

Comment:
Mmm, perhaps something other than `needsTx` here (which is difficult to distinguish from `wantsTx`). What about `offersTx` or `haveTx` or even `toSendTx`?

Reply:
Ah yes I changed it in the tests but not here...